File: fragments/C/errorhandling.c
void *context = zmq-ctx-new ();
assert (context);
void *socket = zmq-socket (context, ZMQ-REP);
assert (socket);
int rc = zmq-bind (socket, "tcp://*:5555");
if (rc != 0) {
    printf ("E: bind failed: %s\n", strerror (errno));
    return -1;
}

File: fragments/Java/errorhandling.java
Context context = ZMQ.context (1);
assert (context != null);
Socket socket = context.socket (ZMQ.REP);
assert (socket != null);
try {
    socket.bind ("tcp://*:5555");
} catch (ZMQException e) {
    System.out.printf ("E: bind failed: %s\n", e.toString ());
    throw e;
}
